sdk/lib/opentelemetry/sdk/resources/auto_detector.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+# Copyright 2019 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require 'opentelemetry/sdk/resources/detectors/google_cloud_platform'
+require 'opentelemetry/sdk/resources/detectors/telemetry'
+
+module OpenTelemetry
+  module SDK
+    module Resources
+      # AutoDetector contains detect class method for running all detectors
+      module AutoDetector
+        extend self
+
+        DETECTORS = [
+          OpenTelemetry::SDK::Resources::Detectors::GoogleCloudPlatform,
+          OpenTelemetry::SDK::Resources::Detectors::Telemetry
+        ].freeze
+
+        def detect
```

Comment:
These detectors are running in serial. Right now it's not a big deal, but if we add additional detectors, especially ones with network calls, this could end up blocking for longer than we'd like. Have you considered parallelizing this at all?

Reply:
I expected this to need to be improved as more detectors are added, but my intention was to provide a starting point to be iterated on.  Hadn't really considered parallelizing it yet.